ultet/src/main/scala/za/co/absa/ultet/model/TransactionGroup.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.ultet.model
+
+
+sealed abstract class TransactionGroup
+
+object TransactionGroup {
+  final case object Database      extends TransactionGroup
```

Comment:
Too many groups.
Enough is to have:
* Role
* Database
* Objects
* Indexes (might be possible to merge with Object maybe, but to be safe, let's keep separated).

Reply:
Hmm okay, when I think about it I agree - I wanted to be granular and explicit, but then sorting would be more complicated and not within one big transaction as planned for these objects